lib/src/signed_video_h26x_sign.c

```diff
@@ -304,13 +300,14 @@ generate_sei_nalu(signed_video_t *self, uint8_t **payload, uint8_t **payload_sig
     reserved_byte |= self->is_golden_sei << 6;
     *payload_ptr++ = reserved_byte;
 
-    size_t optional_written_size = tlv_list_encode_or_get_size(
-        self, optional_document_encoders, num_optional_document_encoders, payload_ptr);
+    size_t optional_written_size =
+        tlv_list_encode_or_get_size(self, optional_tags, num_optional_tags, payload_ptr);
     payload_ptr += optional_written_size;
-    size_t mandatory_written_size = !self->is_golden_sei
-        ? tlv_list_encode_or_get_size(
-              self, mandatory_document_encoders, num_man_doc_encoders, payload_ptr)
-        : 0;
+    size_t mandatory_written_size = 0;
+    if (mandatory_document_size > 0) {
+      mandatory_written_size =
+          tlv_list_encode_or_get_size(self, mandatory_tags, num_mandatory_tags, payload_ptr);
+    }
     size_t written_size = optional_written_size + mandatory_written_size;
     SVI_THROW_IF(written_size == 0, SVI_MEMORY);
     payload_ptr += mandatory_written_size;
```

Comment:
Moving the pointer should be done right after the write operation, that is right after line 309.